src/filter.js

```diff
@@ -41,8 +41,24 @@ const EXCLUDED_MATERIAL_TYPES = [
 ];
 
 export default function (record, earliestCatalogTime) {
+
 	const leader = record.varFields.find(f => f.fieldTag === '_');
-	const materialType = record.materialType.code.trim();
+    let materialType = "";
+
+if (record.materialType === undefined) {
+  	materialType = "temporary";  
+  	return false;
+
+}  else {
+
+      if (record.materialType && record.materialType.code) {
+			materialType = record.materialType.code.trim();    
+      } else {
+        return false;
+      }
+}
```

Comment:
- Wrap this in a function. So the final declaration for `materialType` should be: `const materialType = someFunction();`
- After the declaration of `materialType`, test if it's undefined (`if (!materialType) {`) and return false if it does.
- If the function: You only need the if-statement `if (record.materialType && record.materialType.code) {` and then return the trimmed value. Otherwise the function will return `undefined` by default.